src/server/worker/datum/iterator.go

```diff
@@ -70,22 +70,24 @@ func (ji *jobIterator) Iterate(cb func(*Meta) error) error {
 }
 
 type commitIterator struct {
-	pachClient *client.APIClient
-	commit     *pfs.Commit
-	pathRange  *pfs.PathRange
+	ctx       context.Context
+	c         pfs.APIClient
+	commit    *pfs.Commit
+	pathRange *pfs.PathRange
 }
 
 // NewCommitIterator creates an iterator for the specified commit and repo.
-func NewCommitIterator(pachClient *client.APIClient, commit *pfs.Commit, pathRange *pfs.PathRange) Iterator {
+func NewCommitIterator(ctx context.Context, c pfs.APIClient, commit *pfs.Commit, pathRange *pfs.PathRange) Iterator {
 	return &commitIterator{
-		pachClient: pachClient,
-		commit:     commit,
-		pathRange:  pathRange,
+		ctx:       ctx,
```

Comment:
I should have mentioned this yesterday but the thought didn't occur to me; it would be nice if these contexts were `pctx.Child(ctx, "commitIterator")`.  Just makes the logs extra clear.  It also wouldn't be terrible to `log.SpanContext` around these, when the iterator is done, call the returned `done` callback, and then we can see from the logs how long this iterator lived.  Generally for anything that takes time, a Span around the operation is nice; leaks become clearer, unexpectedly long iterations become clearer, etc.

Reply:
Will do.